.ci/Jenkinsfile

```
@@ -114,7 +119,12 @@ pipeline {
             def agents = readYaml(file: '.ci/.jenkins-agents.yml')
             def parallelTasks = [:]
             agents['agents'].each { agent ->
-              parallelTasks["${agent.REPO}"] = generateStepForAgent(repo: "${agent.REPO}", filesType: "gherkin", filesPath: "${agent.FEATURES_PATH}")
+              if (agent.FEATURES_PATH) {
+                def repository = 'apm-agent-' + agent.REPO
+                if (env.SELECTED_AGENT == 'All' || env.SELECTED_AGENT == repository) {
```

`agentMapping.id` returns the name of the repo without `apm-agent-`, see https://github.com/elastic/apm-pipeline-library/blob/master/vars/agentMapping.groovy#L79
```suggestion
                if (env.SELECTED_AGENT == 'all' || env.SELECTED_AGENT == agent.REPO) {
```

---

Mmm I think we thought it in reverse: the agent.REPO contains the full repository name (i.e. `apm-agent-java`), so the logic should be:
```groovy
def repository = 'apm-agent-' + env.SELECTED_AGENT
if (env.SELECTED_AGENT == 'all' || repository == agent.REPO) {
```
or, we should change the descriptor in the .ci directory